src/ipld/cid_hashmap.rs

```diff
@@ -149,29 +149,32 @@ mod tests {
     use super::*;
     use cid::multihash::{self, MultihashDigest};
     use fvm_ipld_encoding::DAG_CBOR;
+    use quickcheck::Arbitrary;
+    use quickcheck::Gen;
     use quickcheck_macros::quickcheck;
 
-    fn generate_hash_maps(cid_vector: Vec<(Cid, u64)>) -> (CidHashMap<u64>, HashMap<Cid, u64>) {
-        let mut cid_hash_map = CidHashMap::new();
-        let mut hash_map = HashMap::new();
-        for item in cid_vector.iter() {
-            cid_hash_map.insert(item.0, item.1);
-            hash_map.insert(item.0, item.1);
-
-            // Quickcheck does not reliably generate the DAG_CBOR/Blake2b variant of V1 CIDs; need to ensure we have enough samples of this variant in the map for testing, so generate this variant from the values in the key-value pairs.
-            let cid_v1 = Cid::new_v1(
-                DAG_CBOR,
-                multihash::Code::Blake2b256.digest(&item.1.to_be_bytes()),
-            );
-            cid_hash_map.insert(cid_v1, item.1);
-            hash_map.insert(cid_v1, item.1);
+    impl<V> Arbitrary for CidHashMap<V>
+    where
+        V: Arbitrary,
+    {
+        fn arbitrary(g: &mut Gen) -> Self {
+            let cid_vector = Vec::<(Cid, u64)>::arbitrary(g);
+            let mut cid_hash_map = CidHashMap::new();
+            for item in cid_vector.iter() {
+                cid_hash_map.insert(item.0, V::arbitrary(g));
+                // Quickcheck does not reliably generate the DAG_CBOR/Blake2b variant of V1 CIDs; need to ensure we have enough samples of this variant in the map for testing, so generate this variant from the values in the key-value pairs.
```

Comment:
What are the tests that rely on this?

Reply:
All of them. We want a uniform sampling of CID variants, but quickcheck only rarely generates the V1/DAG_CBOR/Blake2b variant.

Reply:
Alright, I don't see a reason to be strict about this. We needed the `Arbitrary` implementation that works and here it is. It's much better than having to call a method.